crates/nu-cli/src/syntax_highlight.rs

```diff
@@ -429,15 +367,14 @@ fn find_matching_block_end_in_expr(
                     None
                 }
             }
+
+            // all other non-nested expressions
+            _ => None,
```

Comment:
The old match with independent arms is certainly unnecessary repetition. A minor downside of this change would be that we have to be careful if we add a new enum variant to `Expr` as this would compile fine without alerting us to another place where we need to handle the new variant.